presto-spark-base/src/main/java/com/facebook/presto/spark/util/PrestoSparkUtils.java

```diff
@@ -271,4 +277,26 @@ protected T computeNext()
             }
         };
     }
+
+    public static PrestoSparkFailure toPrestoSparkFailure(ExecutionFailureInfo executionFailureInfo)
```

Comment:
[major] pass `session` here if you can. If not, it's ok as long as you get my following points:

- test if you are using broadcast join here
- test if you have retry on broadcast join oom enable
- pass `DISABLE_BROADCAST_JOIN` flag into `PrestoSparkFailure`

I haven't checked if there is any dependency or classloader issue. Even that, should be able to workaround accordingly 

BTW, #15857 is the retry design for presto classic. Of course, many of the logics in that PR may not apply here because we were dealing with queueing and client RPCs for the most of the PR. But the error interface is similar.